Could not deploy treasuryDAO because MetadataCarrier contract deployment failed

Steps to reproduce:

1. Create a treasury DAO
2. Launch it
3. Receive following error:

```
mutation.js:103 Error: Could not deploy treasuryDAO because MetadataCarrier contract deployment failed
    at useOriginate.ts:61
    at l (runtime.js:63)
    at Generator._invoke (runtime.js:293)
    at Generator.next (runtime.js:118)
    at r (asyncToGenerator.js:3)
    at s (asyncToGenerator.js:25)
```

--- a/src/modules/creator/components/DeploymentLoader.tsx
+++ b/src/modules/creator/components/DeploymentLoader.tsx
@@ -101,6 +101,7 @@ export const DeploymentLoader: React.FC<Props> = ({
   activeStep,
   error,
 }) => {
+  const errorMessage = "Something went wrong, please try again later."
   const [focusedState, setFocusedState] = useState(0);
   const isFinished = activeStep === states.length;
   const isStarted = Number.isInteger(activeStep);
@@ -125,7 +126,7 @@ export const DeploymentLoader: React.FC<Props> = ({
           <WaitingText variant="subtitle1" color="textSecondary">
             {showActiveText
               ? error
-                ? error.message
+                ? errorMessage
                 : states[focusedState].activeText
               : showCompletedText
               ? states[focusedState].completedText

--- a/src/modules/explorer/pages/Proposals.tsx
+++ b/src/modules/explorer/pages/Proposals.tsx
@@ -126,7 +126,7 @@ export const Proposals: React.FC = () => {
                     onClick={onFlush}
                     disabled={!dao}
                   >
-                    FLUSH
+                    EXECUTE
                   </ViewButton>
                   <Tooltip title="Execute all passed proposals and drop all expired or rejected">
                     <InfoIconInput color="secondary" />

--- a/src/services/contracts/metadataCarrier/deploy.ts
+++ b/src/services/contracts/metadataCarrier/deploy.ts
@@ -65,12 +65,18 @@ export const deployMetadataCarrier = async ({
     const contract = await c.contract();
     console.log("Metadata Carrier deployment completed", c);
     return { contract, keyName, deployAddress: contract.address };
-  } catch (e) {
+  } catch (e: Error | any | unknown) {
     // This should be handled above!
     if (e.name === "UnconfiguredSignerError") {
       // If this happens its because the user is not connected to any wallet
       // Let's connect to a wallet provider and trigger the deployment method again
     }
-    console.log("error ", e);
+    if (e instanceof Error) {
+      const error: Error = e;
+      console.log(error.name + ": " + error.message + "\n" + error.stack);
+    }
+    else {
+      console.log(e);
+    }
   }
 };

